src/github/createPRViewProvider.ts

```diff
@@ -0,0 +1,258 @@
+	}
+
+	private async getTitle(): Promise<string> {
+		const method = vscode.workspace.getConfiguration(PR_SETTINGS_NAMESPACE).get<PullRequestTitleSource>(PR_TITLE, PullRequestTitleSourceEnum.Ask);
+
+		switch (method) {
+
+			case PullRequestTitleSourceEnum.Branch:
+				return this._folderRepositoryManager.repository.state.HEAD!.name!;
+
+			case PullRequestTitleSourceEnum.Commit:
+				return titleAndBodyFrom(await this._folderRepositoryManager.getHeadCommitMessage()).title;
+
+			default:
+				return '';
+		}
+	}
+
+	private async getDescription(): Promise<string> {
+		const method = vscode.workspace.getConfiguration('githubPullRequests').get<PullRequestDescriptionSource>('pullRequestDescription', PullRequestDescriptionSourceEnum.Ask);
+
+		switch (method) {
+
+			case PullRequestDescriptionSourceEnum.Template:
+				const templateUris = await this._folderRepositoryManager.getPullRequestTemplates();
+				if (templateUris[0]) {
+					try {
+						const templateContent = await vscode.workspace.fs.readFile(templateUris[0]);
+						return templateContent.toString();
+					} catch (e) {
+						// Logger.appendLine(`Reading pull request template failed: ${e}`);
+						return '';
+					}
+				}
+
+			case PullRequestDescriptionSourceEnum.Commit:
+				return titleAndBodyFrom(await this._folderRepositoryManager.getHeadCommitMessage()).body;
+
+			default:
+				return '';
+		}
+	}
+
+	public async initializeParams(): Promise<void> {
+		const pullRequestDefaults = await this._folderRepositoryManager.getPullRequestDefaults();
+
+		const defaultRemote: RemoteInfo = {
+			owner: pullRequestDefaults.owner,
+			repositoryName: pullRequestDefaults.repo
+		};
+
+		Promise.all([
+			this._folderRepositoryManager.getGitHubRemotes(),
+			this._folderRepositoryManager.listBranches(pullRequestDefaults.owner, pullRequestDefaults.repo),
+			this.getTitle(),
+			this.getDescription()
+
+		]).then(result => {
+			const [githubRemotes, branchesForRemote, defaultTitle, defaultDescription] = result;
+
+			const remotes: RemoteInfo[] = githubRemotes.map(remote => {
+				return {
+					owner: remote.owner,
+					repositoryName: remote.repositoryName
+				};
+			});
+
+			this._postMessage({
+				command: 'pr.initialize',
+				params: {
+					availableRemotes: remotes,
+					defaultRemote,
+					defaultBranch: pullRequestDefaults.base,
+					branchesForRemote,
+					defaultTitle,
+					defaultDescription
+				}
+			});
+		});
+	}
+
+	private async askForAlwaysUse(stateKey: string, message: string, titleSource: PullRequestTitleSourceEnum): Promise<void> {
+		const config = vscode.workspace.getConfiguration(PR_SETTINGS_NAMESPACE);
+		if (config.get(PR_TITLE) === titleSource) {
+			return;
+		}
+
+		const showPrompt = PersistentState.fetch('prompts', stateKey);
+		if (!showPrompt || showPrompt === PersistentState.MISSING) {
+			vscode.window.showInformationMessage(
+				`Would you like to always use the ${message} as the title of the pull request?`,
+				{ modal: true },
```

Comment:
Generally I think modal dialogs should only be used when action must be taken by the user. Since this is kind of an optional think, maybe it would be better as non-modal?

Reply:
I think in this case it makes more sense to have it as a modal, because I think it would be easy to miss as a notification, and also this matches some other patterns we have, like when you move a ts file for the first time and you get a dialog asking if you always want to rename imports. I'll bring this up in the UX call, though